### DISCUSSION TITLE
Give tiles an 'override physics on lower layers' boolean property.

### DISCUSSION THREAD

**Discussion by Proggle** [godotengine/godot-proposals#7754] [2D]
Here's the problem:

A common and natural way to do mapping is something like this:
You draw a river on your map
![image](https://github.com/godotengine/godot-proposals/assets/89282845/2a213f68-18f8-4830-88ff-f90fbbe04909)
it has collision, preventing the player from walking over it
![image](https://github.com/godotengine/godot-proposals/assets/89282845/c2a5d61f-ed32-4ec7-8e83-9d526bea29d0)
You want to make it crossable, so you put in a bridge you designed which has appropriate collision
![image](https://github.com/godotengine/godot-proposals/assets/89282845/b1ef419e-24ba-4087-a6d6-59c495f8be84)

Unfortunately, in Godot, collision is strictly additive, so you end up with this
![image](https://github.com/godotengine/godot-proposals/assets/89282845/a15de37d-98cf-4d9d-bd76-e5d0b5de77a0)
instead of this
![image](https://github.com/godotengine/godot-proposals/assets/89282845/c5c30777-af6d-4955-bbd3-30d43d77bffe)

This is a big enough problem for workflow that multiple people have proposed and implemented methods for turning off physics on a particular layer, ( see https://github.com/godotengine/godot-proposals/issues/6881 https://github.com/godotengine/godot-proposals/issues/4236 https://github.com/godotengine/godot-proposals/issues/3092 ), which have been rejected since this puts too much control into the TileMap compared to the TileSet (per groud, "we have to draw a limit to what belongs to the TileSet and what belong to the TileMap.").  Nevertheless, people keep wanting an easy solution so they can get back to mapping.

The current methods for working around this additive physics situation are:

A) make duplicate versions of all your tiles that might be underneath something, then map those alternate versions in under your bridge after placing it
B) leave a hole in your TileMap, put in a second TileMap underneath it with no physics, and map everything under it
C) attach a script to your TileMap that, at runtime, dynamically changes physics of underlying tiles to remove them

All of these have problems (such as breaking the visible collisions feature in the map editor), and are time consuming to do as a dev.  B and C also are a TileMap-based solution instead of having this property decision belong to the TileSet, which isn't ideal

I think there's room for a better solution that's based on a property in the TileSet, which would make it easy to just plop down bridges wherever you needed to, and have them work the way you want without needing to alter your node structure, create new tiles, or code in special runtime exception.

Tiles in the TileSet should have override_lower_layers boolean property.  If this is set to "True", any other tiles in a TileMap with the same x and y coordinates which is on a lower layer will not create physics colliders.  

(In the case where multiple tiles have this flag, the one on a higher z-level will blank colliders of everything below it regardless of whether the flag is set.)

I think there are several distinct advantages of handling this problem this way.  First, it keeps the solution in the TileSet and makes the behavior universal across all TileMaps, keeping tile properties where they belong.. Second, since it is not calculated at runtime, you can still preview the collision of your map in the editor, making it easy to test what can and can't be walked on without having to rub your character all over every object.  Third, this is by far the easiest way to solve this specific use case - if you put a bridge with this property onto _any_ TileMap the player will immediately be able to cross it, no problem, no special preparations needed.  And if you later remove the bridge, it won't leave behind an empty walkable patch of river that you might forget to fix.

**Comment by DinoMC** [comment]
Sorry to answer an old topic but I'm facing this issue right now and making so many duplicate tiles to handle this is killing me, and this is the only relevant link I've been able to find.

Do you know if there has been any news about this since, or an existing issue I wasn't able to find?

Thanks

**Comment by AThousandShips** [comment]
Note that TileMap is now deprecated and new projects are directed to use `TileMapLayer` in 4.3, so changes like this are unlikely to happen to this type
